concent_api/core/payments/storage.py

```diff
@@ -0,0 +1,145 @@
+from core.models import PendingEthereumTransaction
+from core.payments import service as payments_service
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseTransactionsStorage():
+    """
+    Concent custom implementation of TransactionsStorage interface used to store Ethereum transaction data into
+    database using Django models.
+    """
+
+    def __init__(self, *args: List, **kwargs: Dict) -> None:
+        super().__init__(*args, **kwargs)
+
+        if not GlobalTransactionState.objects.filter(pk=0).exists():
+            global_transaction_state = GlobalTransactionState(
+                pk=0,
+                nonce=payments_service.get_transaction_count(),  # pylint: disable=no-value-for-parameter
+            )
+            global_transaction_state.full_clean()
+            global_transaction_state.save()
+
+    @transaction.atomic(using='control')
+    def get_nonce(self) -> int:
+        """
+        Returns the nonce for the next transaction.
+        """
+        global_transaction_state = self._get_locked_global_transaction_state()
+        return global_transaction_state.nonce
+
+    @transaction.atomic(using='control')  # pylint: disable=no-self-use
+    def get_all_tx(self) -> List[Transaction]:
+        """
+        Returns the list of all transactions.
+        """
+        return [
+            Transaction(
+                nonce    = int(ethereum_transaction.nonce),
+                gasprice = int(ethereum_transaction.gasprice),
+                startgas = int(ethereum_transaction.startgas),
+                value    = int(ethereum_transaction.value),
+                v        = int(ethereum_transaction.v),
+                r        = int(ethereum_transaction.r),
+                s        = int(ethereum_transaction.s),
+                to       = encode_hex(ethereum_transaction.to.tobytes()),
+                data     = ethereum_transaction.data.tobytes(),
+            )
+            for ethereum_transaction in PendingEthereumTransaction.objects.all()
+        ]
+
+    @transaction.atomic(using='control')
+    def put_tx_and_inc_nonce(self, tx: Transaction) -> None:
+        """
+        Save a valid transaction and increase the nonce.
+        """
+        global_transaction_state = self._get_locked_global_transaction_state()
+
+        if global_transaction_state.nonce != tx.nonce:
+            raise Exception(
+                f'Transaction nonce does not match. '
+                f'Current={global_transaction_state.nonce}, '
+                f'Transaction={tx.nonce}'
+            )
+
+        pending_ethereum_transaction = PendingEthereumTransaction(
+            nonce       = tx.nonce,
+            gasprice    = tx.gasprice,
+            startgas    = tx.startgas,
+            value       = tx.value,
+            v           = tx.v,
+            r           = tx.r,
+            s           = tx.s,
+            data        = tx.data,
+            to          = tx.to,
```

Comment:
Style:)

Reply:
Same here, done for readability, but if you or anyone insist I will change it.

Reply:
For me is okay I didn't want to change our style and for me it is more readable :) I have just pointed this because it's not matched with our current style